File: Java/GuessGame.java
// Grant Commodore

/* This program generates a random number between 0 and 100.  Then, it allows the user 10 guesses.
If the guess is correct, "You guessed it! The number was" is printed along with the number. If 
the guess was too high or too low, the user is notified and another guess is accepted .If the 
user does not guess in 10 guesses, "you lose, the number was" is printed along with the number. */
import java.util.Scanner;

public class GuessGame {

    // Method that generates a random number between 0 and 100.
    public static int generateRandom() {
        return (int) (Math.random() * 100);
    }

    // Method that checks if the guess is correct.
    public static boolean checkGuess(int guess, int random, int numGuesses) {
        // If the guess is correct, print "You guessed it! The number was" is printed
        // along with the number.
        if (guess == random) {
            System.out.println("You guessed it! The number was " + random + ".");
            return true;
            // If the guess was too high, print "Too high, try again."
        } else if (guess > random && numGuesses < 10) {
            System.out.println("Too high, try again.");
            return false;
            // If the guess was too low, print "Too low, try again."
        } else if (guess < random && numGuesses < 10) {
            System.out.println("Too low, try again.");
            return false;
            // If the user does not guess in 10 guesses, print "you lose, the number was" is
            // printed along with the number.
        } else {
            System.out.println("You lose, the number was " + random + ".");
            return false;
        }
    }

    // Main method that begins the execution of the Guessing Game.
    public static void main(String[] args) {
        int random = generateRandom();
        int guess = 0;
        int guessCount = 1;
        System.out.println("Welcome to the Guess Game!");
        System.out.println("You have 10 guesses to guess the random number between 0 and 100.");
        System.out.println("--------------------------------------------------------------------------------");
        Scanner input = new Scanner(System.in);
        // While loop that continues to run until the user guesses the correct number or
        // the user runs out of guesses.
        while (guessCount <= 10) {
            System.out.print("Enter a guess: ");
            guess = input.nextInt();
            if (checkGuess(guess, random, guessCount)) {
                break;
            }
            guessCount++;
        }

        // Close the Scanner.
        input.close();
    }

}

File: python/guess_game.py
"""
Author: Grant Commodore

This program generates a random number between 0 and 100.  Then, it allows the user 10 guesses.
If the guess is correct, "You guessed it! The number was" is printed along with the number. If 
the guess was too high or too low, the user is notified and another guess is accepted .If the 
user does not guess in 10 guesses, "you lose, the number was" is printed along with the number.
"""

import random

# Generate a random number between 0 and 100.
def get_random_num():
    random_num = random.randint(0, 100)
    return random_num

# Get the user's guess.
def get_guess():
    guess = int(input("Enter your guess: "))
    return guess

# Determine if the guess is correct.
def is_correct(guess, correct_num, count):
    # If this is the last guess and the guess is not correct, return false;
    if (count == 10) and (guess != correct_num):
        print("You lose! The number was", correct_num, ".")
        print("")
        return False
    # If the guess is correct, return true.
    elif guess == correct_num:
        print  ("You guessed it! The number was", correct_num, ".")
        print("")
        return True
    # If the guess is too high, return false.
    elif guess < correct_num:
        print("Too low, try again.")
        print("")
        return False
    # If the guess is too low, return false.
    else:
        print("Too high, try again.")
        print("")
        return False

# Main program
def main():
    # Get the random number.
    correct_num = get_random_num()

    # Give the user 10 guesses.
    for i in range(10):
        guess = get_guess()
        guess_is_correct = is_correct(guess, correct_num, i)
        # If the guess is correct, end the program.
        if guess_is_correct == True:
            break
    if (guess_is_correct == False):
        print("You lose! The number was", correct_num, ".")
        print("")
main()